components/infobox/wikis/mobilelegends/infobox_player_custom.lua

```diff
@@ -0,0 +1,97 @@
+---
+-- @Liquipedia
+-- wiki=mobilelegends
+-- page=Module:Infobox/Person/Player
+--
+-- Please see https://github.com/Liquipedia/Lua-Modules to contribute
+--
+
+local Player = require('Module:Infobox/Person')
+local String = require('Module:StringUtils')
+local Class = require('Module:Class')
+local Earnings = require('Module:Earnings')
+local TeamHistoryAuto = require('Module:TeamHistoryAuto')
+local Role = require('Module:Role')
+local Region = require('Module:Region')
+
+local Injector = require('Module:Infobox/Widget/Injector')
+local Cell = require('Module:Infobox/Widget/Cell')
+local Title = require('Module:Infobox/Widget/Title')
+local Center = require('Module:Infobox/Widget/Center')
+
+local _pagename = mw.title.getCurrentTitle().prefixedText
+local _role
+local _role2
+local _EMPTY_AUTO_HISTORY = '<table style="width:100%;text-align:left"></table>'
+
+local CustomPlayer = Class.new()
+
+local CustomInjector = Class.new(Injector)
+
+local _args
+
+function CustomPlayer.run(frame)
+	local player = Player(frame)
+	_args = player.args
+
+	player.calculateEarnings = CustomPlayer.calculateEarnings
+	player.adjustLPDB = CustomPlayer.adjustLPDB
+	player.createWidgetInjector = CustomPlayer.createWidgetInjector
+
+	return player:createInfobox(frame)
+end
+
+function CustomInjector:parse(id, widgets)
+	if id == 'history' then
+		local manualHistory = _args.history
+		local automatedHistory = TeamHistoryAuto._results({
+			convertrole = 'true',
+			player = _pagename
+		}) or ''
+		automatedHistory = tostring(automatedHistory)
+		if automatedHistory == _EMPTY_AUTO_HISTORY then
+			automatedHistory = nil
+		end
+
+		if not (String.isEmpty(manualHistory) and String.isEmpty(automatedHistory)) then
+			return {
+				Title{name = 'History'},
+				Center{content = {manualHistory}},
+				Center{content = {automatedHistory}},
+			}
+		end
+	elseif id == 'region' then return {}
+	elseif id == 'role' then
+		_role = Role.run({role = _args.role})
+		_role2 = Role.run({role = _args.role2})
+		return {
+			Cell{name = 'Role(s)', content = {_role.display, _role2.display}}
+		}
+	end
```

Comment:
Why the need to override? Shouldn't it be done in commons?

Reply:
the commons one (due to RL) only displays the role, no role 2 and no usage of a module:Role
```lua

		Customizable{id = 'role', children = {
			Cell{name = 'Role', content = { args.role }
				}
			}
		},
```

Reply:
Should like we should decide on a standardised usage that works for all wikis (in the long run) and have RL run a custom until they have it setup?

Reply:
feel free to open a pr for it into the commons module
(for sc2 i overwrite any role implementation as we do not display it at all)